app/src/test/resources/design-document.md

```diff
@@ -0,0 +1,118 @@
+
+## Use Cases
+u1. _As a user, I would like to be able to view the books in my collection_
+u2. _As a user, I would like to favorite the books I like most_
+U3. _As a user, I would like to view to store the price I paid for such book_
+U5. _As a user, I would like to update my collection with new books or collectibles I have purchased_
+U6. _As a user, I would like to update my collection, to get rid of books I might have sold off and no longer own_
+U7. _As a user, I would like add a book to a new series I have started collecting_
+U8. _As a user, I would like create an account_
+U9 _As a user, I would like to login to my account_
+U10. _As a user, I would like to see collection based of my login_
+
+Stretch goals:
+- Search Option
+- Adding Images to the comics added(if wanted, user takes a picture and able to store that image)
+- Better design on website
+
+## Project Scope
+- view books that have been added to my collection
+- add books to a new series I am collection
+- view all books in my collection
+
+## Proposed Architecture Overview
+We will use API Gateway and Lambda to create endpoints()
+(GetAllComicBooksLambda, GetAllFavoriteLambda, GetAllStampsLambda, GetAllCoinsLambda, AddComicBookToCollectionLambda, AddStampsToCollectionLambda, AddCoinsToCollectionLambda, UpdateComicBookCollectionLambda, UpdateStampCollectionLambda, UpdateCoinCollectionLambda, SearchCollectionLambda)
+_that will handle the creation, update, and retrieval of events and vendors to satisfy our requirements_
+
+## API
+
+## 6.1 Public Models
+```
+// ComicBookModel
+
+String titleId
+String volumeNumber
+String issueNumber
+String date
+Int price
+Boolean isFavorite
+String publisher
+```
+
+## 6.2 _Get All Comic Books Endpoint_
+
+_* Accepts `GET` requests to `/comicbooks/`
+* Scans comics table and returns all ComicBookModels.
+    * If no ComicBookTable is found
+      `ComicBookNotFoundException`_
+
+
+## 6.3 _Get One ComicBook Endpoint_
+
+_* Accepts `GET` requests to `/comicbooks/:titleId/volumeNumber`
+* Accepts a comicbook ID and volumeNumber returns the corresponding Comic Book Title.
+    * If the given comicBook ID is not found or volumeNumber is not found, will throw a
+      `ComicBookNotFoundException`_
+
+## 6.4 _Add ComicBook Endpoint_
+
+* Accepts `POST` requests to `/comicbook/titleId/volumeNumber/add`
+* Accepts a comicbook ID and a ComicBookModel to be added.
+    * If the comicBook can not be added to the comicBookTable, will throw an `UnableToAddToTableException`
+
+
+## 6.5 _Update ComicBook Endpoint_
+
+* Accepts `PUT` requests to, `/comicbook/:titleId/volumenumber`
+    * If the given comicbook ID is not found and volumenumber, will throw a
+      `ComicBookNotFoundException`_
+
+## 6.6 _Get All Series Endpoint_
+
+_* Accepts `GET` requests to `/comicbooks/series`
+* Scans series table and returns all Series of pre-populated comic book model.
+    * If no ComicBookTable is found
+      `ComicBookNotFoundException`_
```

Comment:
Would this endpoint return a list of a second public model representing a Series? (vs ComicBookModels)

If so, using '/series' as the path would be more consistent with restful api naming conventions.

Reply:
Yes, Series would be a better naming convention and not using the comicbooks one